src/storage/storages.rs

```diff
@@ -174,13 +175,13 @@ unsafe impl<T> DistinctStorage for NullStorage<T> {}
 pub struct VecStorage<T>(Vec<T>);
 
 impl<T> UnprotectedStorage<T> for VecStorage<T> {
-    unsafe fn clean<F>(&mut self, has: F)
+    unsafe fn clean<B>(&mut self, has: B)
     where
-        F: Fn(Index) -> bool,
+        B: BitSetLike,
     {
         use std::ptr;
         for (i, v) in self.0.iter_mut().enumerate() {
```

Comment:
I'm unsure if this is faster than iterating over the bitset itself.

Reply:
It depends

Reply:
I'm sorry, but this answer really doesn't help. Could you elaborate, please?

Reply:
If most entities has the component - enumerating and check may be faster than iterating over the bitset. Otherwise it must be faster to iterate over bitset.

Reply:
Anyway. Changing nothing can't make things to worse.